### PR TITLE
Automated documentation + Github community guidelines + Pip installable + Versioning

### PR DESCRIPTION
**Pull request by ayush9pandey**
This is a pull request that would update the biocrnpyler version to 1.0.0 (and so should be merged once all the pending changes have been completed and we are ready to merge to master). Along with this pull request, I am also adding a "contributing.md" file inside the docs directory that would serve as the guideline page for code contributions to biocrnpyler. Also, this PR will do all the setting-up work required for sphinx documentation and pip installation. Please feel free to comment on things to do here.

### CODE REVIEW

**Comment by ayush9pandey** [comment]
The README file also needs an update, that I haven't done yet. If someone wants to jump in, please feel free to do so with a PR to this PR.

**Comment by WilliamIX** [comment]
We should make sure to test the pip installation on linux, mac, and windows.

**Comment by zoltuz** [comment]
@ayush9pandey just a small change: contributing.md should be all caps (CONTRIBUTING.md) and it should in the project root, so GitHub can recognize it.

**Comment by ayush9pandey** [comment]
> 
> 
> @ayush9pandey just a small change: contributing.md should be all caps (CONTRIBUTING.md) and it should in the project root, so GitHub can recognize it.

I think I made all caps, it shows up like that on my computer anyways. I followed instructions from Github here: https://docs.github.com/en/github/building-a-strong-community/setting-guidelines-for-repository-contributors it says here that the file needs to be in docs/?

**Comment by ayush9pandey** [comment]
The package is available on Test PyPi here: https://test.pypi.org/project/biocrnpyler/ 

I am inclined to remove bokeh and fa2 as package installation dependencies because pip collects those to make sure the installation goes through correctly (this is different from how setuptools worked where these were not collected unless the user wanted to). Let me know your thoughts.

**Comment by zoltuz** [comment]
> The package is available on Test PyPi here: https://test.pypi.org/project/biocrnpyler/
> (once everything is finalized I will move to the actual PyPi).
> I am inclined to remove bokeh and fa2 as package installation dependencies because pip collects those to make sure the installation goes through correctly (this is different from how setuptools worked where these were not collected unless the user wanted to). Let me know your thoughts.

The less official dependencies we start with, the less problem we have to take care of. There was already a dependency problem with Bokeh (#106 ), I bet more is waiting for us just around the corner (for example, for every dependency problem, we have to release a package with a new version at PyPi).

This is what we have now as __offiial__ dependencies:

 - [X] python-libsbml
 - [X] numpy
 - [ ] matplotlib<=3.2.2',
 - [X] networkx,
 - [ ] bokeh==1.4.0,
 - [ ] fa2

I suggest to take the ticked ones (python-libsmbl, numpy, networkx as core dependencies) and move the rest into a ```plotting_requirements.txt```, which one can install if needed:
```
pip install biocrnpyler
pip install -r plotting_requirement.txt
```

**Comment by zoltuz** [comment]
I've found a more elegant solution: Networkx does this: https://github.com/networkx/networkx/blob/master/setup.py#L127

to install the dependencies:
```
pip install biocrnpyler
```
to install all optional dependencies:
```
pip install biocrnpyler[all]

I quite like this idea, any thoughts?

**Comment by murrayrm** [comment]
Is there a reason you are pinning requirements to specific version?  Especially for plotting packages that don't seem to be core to the functionality of BioCRNpyler?

**Comment by ayush9pandey** [comment]
> 
> 
> Is there a reason you are pinning requirements to specific version? Especially for plotting packages that don't seem to be core to the functionality of BioCRNpyler?

We decided to move all plotting libraries as optional dependencies. Also, the versions for some of those have to be pinned to be greater than equal to a certain version number due to the features that have been used.

**Comment by ayush9pandey** [comment]
The latest version on Test PyPi has the optional dependencies installation now working: https://test.pypi.org/project/biocrnpyler/

@WilliamIX : Please let me know once you have merged all other PRs so that I can get this one ready to be merged.

**Comment by ayush9pandey** [comment]
The alpha release (version 0.2.1) is available on pip: `pip install biocrnpyler==0.2.1` should install that version. 

As soon as this gets merged to master, I will upload the latest version on pip as well.

File: biocrnpyler/plotting.py [hunk 1]
```diff
@@ -6,18 +6,22 @@
...
-from bokeh.models import (BoxSelectTool, Circle,Square, EdgesAndLinkedNodes, HoverTool,
-                          MultiLine, NodesAndLinkedEdges, Plot, Range1d, TapTool,PanTool,WheelZoomTool)
-from bokeh.palettes import Spectral4
-from bokeh.models.graphs import from_networkx
-from fa2 import ForceAtlas2
 import numpy as np
 from .propensities import MassAction, Hill
 from matplotlib import cm
-import matplotlib.pyplot as plt
+import matplotlib.pyplot as plthttps://test.pypi.org/project/biocrnpyler/1.0.2/
```
Comment (zoltuz) [inline biocrnpyler/plotting.py:24]: something went wrong here

**Pull request merged**